src/Data/ProgrammesDb/EntityRepository/CoreEntityRepository.php

```diff
@@ -196,6 +196,30 @@ public function findByPidFull(string $pid, string $entityType = 'CoreEntity'): ?
         return $this->resolveCategories($withHydratedParents)[0];
     }
 
+    /**
+     * @param string[] $pids
+     * @param string $entityType
+     * @return array
+     */
+    public function findByPids(array $pids, string $entityType = 'CoreEntity'): array
+    {
+        $this->assertEntityType($entityType, self::ALL_VALID_ENTITY_TYPES);
+        $qb = $this->getEntityManager()->createQueryBuilder()
+            ->select(['entity', 'image', 'masterBrand', 'network', 'mbImage', 'category', 'nwImage'])
+            ->from('ProgrammesPagesService:' . $entityType, 'entity') // For filtering on type
+            ->leftJoin('entity.image', 'image')
+            ->leftJoin('entity.masterBrand', 'masterBrand')
+            ->leftJoin('masterBrand.image', 'mbImage')
+            ->leftJoin('masterBrand.network', 'network')
+            ->leftJoin('network.image', 'nwImage')
+            ->leftJoin('entity.categories', 'category')
+            ->andWhere('entity.pid IN (:pids)')
+            ->setParameter('pids', $pids);
```

Comment:
What if the array contains something other that strings?

Reply:
This function is for internal use in pages service, the input should be checked in the Service call, I'm reluctant to duplicate that